### PR TITLE
Completed Day 7 - Please Review

### PR DESCRIPTION
**Pull request by bharatreddypothireddy**
Not a correct solution . Did not execute for all test cases on leet code. Confused as to how to approach solving it.

### CODE REVIEW

File: HouseRobber.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// Time Complexity :O(n) --> beacuse of for loop
+// Space Complexity :O(1)  ---> because of Hashmap
+// Did this code successfully run on Leetcode : No
+// Any problem you faced while coding this : Didn'' understand how to approach for different test cases
+
+
+// Your code here along with comments explaining your approach
+
+class Solution {
+    public int rob(int[] nums) {
```
Comment (super30admin) [inline HouseRobber.java]: Its not the case of adding even and odd.
You need to where to go (n+2 or n+3) whichever is maximum out of these two.

File: CoinChange.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// Time Complexity :O(n*n) --> beacuse of for loop
+// Space Complexity :O(n)  ---> because of Hashmap
+// Did this code successfully run on Leetcode : No
+// Any problem you faced while coding this : Didn'' understand how to approach for different test cases
+
+
```
Comment (super30admin) [inline CoinChange.java]: This Solution wont work.
Try once again using Knapsack approcah.

File: HouseRobber.java (hunk 1 above)
Comment (bharatreddypothireddy) [reply HouseRobber.java]: Completed using dynamic programming approach by finding the maximum amount a robber can rob on reaching a particular house at index 'i' and adding them to a dp array . Finally return the last value of the array.

File: CoinChange.java (hunk 2 above)
Comment (bharatreddypothireddy) [reply CoinChange.java]: Completed using tabulation method approach taught by Jaspinder.

**Comment by jaspindersingh83** [comment]
Good

**Pull request closed without merging**